gridstatus/ercot.py

```diff
@@ -491,6 +845,73 @@ def _handle_html_data(self, df, columns):
         cols_to_keep = ["Time"] + list(columns.keys())
         return df[cols_to_keep].rename(columns=columns)
 
+    def _filter_by_settlement_point_type(self, df, location_type):
+        """Filter by settlement point type"""
+        norm_location_type = location_type.upper()
+        if norm_location_type == LOCATION_TYPE_NODE:
+            df = df[df["SettlementPointType"].isin(RESOURCE_NODE_SETTLEMENT_TYPES)]
+        elif norm_location_type == LOCATION_TYPE_ZONE:
+            df = df[df["SettlementPointType"].isin(LOAD_ZONE_SETTLEMENT_TYPES)]
+        elif norm_location_type == LOCATION_TYPE_HUB:
+            df = df[df["SettlementPointType"].isin(HUB_SETTLEMENT_TYPES)]
+        else:
+            raise ValueError(f"Invalid location_type: {location_type}")
+        return df
+
+    def _filter_by_locations(self, df, field_name, locations):
```

Comment:
there is an existing method `utils.py::filter_lmp_locations` for this operation

rather than pass field name as a parameter I would just do the rename to "Location" before calling since that is the standard name across grid status